etc/systemd/system/zfs-import-scan.service.in

```diff
@@ -4,6 +4,7 @@ DefaultDependencies=no
 Requires=systemd-udev-settle.service
 After=systemd-udev-settle.service
 After=cryptsetup.target
+After=zfs-import-cache.service
 ConditionPathExists=!@sysconfdir@/zfs/zpool.cache
```

Comment:
ConditionPathExists prevents the services from running at the same time.

zfs-import-cache only runs if @sysconfdir@/zfs/zpool.cache exists
zfs-import-scan only runs if @sysconfdir@/zfs/zpool.cache doesn't exist